ethportal-peertest/src/scenarios.rs

```diff
@@ -0,0 +1,42 @@
+use crate::jsonrpc::{
+    make_ipc_request, validate_portal_offer, JsonRpcRequest, HISTORY_CONTENT_KEY,
+    HISTORY_CONTENT_VALUE,
+};
+use crate::{Peertest, PeertestConfig};
+use log::info;
+use serde_json::{json, Value};
+use trin_core::jsonrpc::types::Params;
+
+pub fn test_offer_accept(peertest_config: PeertestConfig, peertest: &Peertest) {
+    info!("Testing OFFER/ACCEPT flow");
+
+    // Store content to offer in the testnode db
+    let store_request = JsonRpcRequest {
+        method: "portal_historyStore".to_string(),
+        id: 11,
+        params: Params::Array(vec![
+            Value::String(HISTORY_CONTENT_KEY.to_string()),
+            Value::String(HISTORY_CONTENT_VALUE.to_string()),
+        ]),
+    };
+
+    let store_result = make_ipc_request(&peertest_config.target_ipc_path, &store_request).unwrap();
+    assert_eq!(store_result.as_str().unwrap(), "true");
+
+    // Send offer request
+    let offer_request = JsonRpcRequest {
+        method: "portal_historyOffer".to_string(),
+        id: 11,
+        params: Params::Array(vec![
+            Value::String(peertest.bootnode.enr.to_base64()),
+            Value::Array(vec![json!(HISTORY_CONTENT_KEY)]),
+        ]),
+    };
+
+    let accept = make_ipc_request(&peertest_config.target_ipc_path, &offer_request).unwrap();
+
+    // Check accept response
+    validate_portal_offer(&accept, peertest);
+
+    // TODO: Check if the stored content item in bootnode's DB matches the offered
```

Comment:
This is blocked by https://github.com/ethereum/trin/pull/277.

Reply:
Why? Is it to make sure every test starts with an empty database?

Reply:
> Why? Is it to make sure every test starts with an empty database?

Yeah, on OFFER/ACCEPT the receiving node stores the content and it seems we don't clean the peertest databases after all tests are done.